### PR TITLE
riscv/syscall: Simplify system call and task startup logic

### PR DESCRIPTION
**Pull request by pussuw**
## Summary
Port the simplifications from ARM64:
- Remove the ugly inline assembly trampoline "do_syscall" and replace it with a simple table lookup and call via function pointer.
- Remove the 2 reserved system calls for up_task_start and up_pthread_start and replace them with an ASM snippet. The result removes an unnecessary ecall from the process startup logic, as well as ensures the stacks are FULLY unwound when the user process starts.
- Remove riscv_current_ksp as obsolete, as now the kernel stack is fully unwound upon exception exit, and we don't need the (hack) riscv_current_ksp anymore.
## Impact
Replace unnecessarily complex code from user system call handling and task startup logic. Only the RISC-V platform is affected. Only memory protected modes (BUILD_PROTECTED/BUILD_KERNEL) are affected.
## Testing
rv-virt:knsh64
rv-virt:ksmp64
rv-virt:pnsh64
Downlink MPFS target with BUILD_KERNEL and 1000+ processes / threads running, starting, exiting all the time

The changes are a backport from ARM64 where the simplified logic was first introduced (by me).

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



This PR appears to meet the NuttX requirements. Here's a breakdown:

**Strengths:**

* **Clear Summary:**  The summary effectively outlines the "why," "what," and "how" of the changes.
* **Specific Impact:**  The impact section clearly identifies the affected platform (RISC-V), build modes (protected/kernel), and the nature of the changes (simplification, not new features).
* **Detailed Testing:**  Testing details include specific targets (rv-virt configurations) and a real-world use case scenario (MPFS target with heavy process/thread load).

**Possible Improvements:**

* **Before/After Logs:** While you mention testing logs, providing snippets of the logs (even if concise) to demonstrate the improvements would strengthen the PR.
* **Issue References:** If this PR addresses a specific issue in the NuttX or NuttX Apps repositories, include the issue numbers for better tracking and context.
* **Documentation:** If these changes impact user-facing behavior or require documentation updates, mention that explicitly. 

**Overall:**

The PR demonstrates a good understanding of the NuttX requirements.  Adding the before/after logs and addressing the minor points above will further enhance its quality.

File: arch/risc-v/src/common/riscv_initialstate.c [hunk 1]
```diff
@@ -169,4 +169,8 @@ void up_initial_state(struct tcb_s *tcb)
 #ifdef CONFIG_ARCH_RISCV_INTXCPT_EXTENSIONS
   riscv_initial_extctx_state(tcb);
 #endif
+
+#ifndef CONFIG_BUILD_FLAT
+  tcb->xcp.initregs   = tcb->xcp.regs;
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_initialstate.c:174]:
```
tcb->xcp.initregs = tcb->xcp.regs;
```

File: arch/risc-v/src/common/riscv_exception_common.S [hunk 2]
```diff
@@ -273,6 +273,42 @@ return_from_exception:
 
   ERET
 
+/****************************************************************************
+ * Function: arm64_jump_to_user
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_exception_common.S:277]: riscv_jump_to_user

File: arch/risc-v/src/common/riscv_exception_common.S [hunk 3]
```diff
@@ -273,6 +273,42 @@ return_from_exception:
...
+ *
+ * Description:
+ *  Routine to jump to user space, called when a user process is started and
+ *  the kernel is ready to give control to the user task in user space.
+ *
+ * arm64_jump_to_user(entry, x0, x1, regs)
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_exception_common.S:283]: riscv_jump_to_user

**Review by acassis — APPROVED** [review]

**Pull request merged**